### PR TITLE
Unify logging guides

### PR DESCRIPTION
**Pull request by MatKuhr**
This PR unifies the two guides on logging with Java. Furthermore it expands the sections on SLF4J and restructures a bit.

For transparency: I already synced offline on merging these two guides with @artemkovalyov but didn't get around to do it earlier.

### CODE REVIEW

File: docs/java/guides/increase-log-level.md [hunk 1]
```diff
@@ -1,88 +0,0 @@
...
-
-## When running app/debugging on localhost ##
-
-For applications that run on localhost you can influence the log level settings via Maven.
-
-### Change Default Log Level
-
-To change the default log level, add this argument to your `mvn` run command:
-```makefile
--Dorg.slf4j.simpleLogger.defaultLogLevel=debug
```
Comment (MatKuhr) [inline docs/java/guides/increase-log-level.md:26]: This part was actually incorrect, it only works if one is using the `slf4j-simple` implementation. But in our archetypes we ship Logback, so usually that won't be the case. Hence removing it.

File: sidebars.js [hunk 2]
```diff
@@ -79,13 +79,12 @@ module.exports = {
         type: 'category',
         label: 'Guides',
         items: [
-          'java/guides/sap-cloud-sdk-linux-how-to',
```
Comment (MatKuhr) [inline sidebars.js:82]: I reordered a bit according to the frequency with which the information would probably be accessed.

File: docs/java/guides/logging-overview.md [hunk 3]
```diff
@@ -3,36 +3,88 @@ id: logging-overview
...
+In order to provide a logging framework you have to add it to the dependency tree.
+Which artifacts are to be added exactly depends on the framework.
+
+To take Logback as an example:
+
+```xml
+<dependency>
+	<groupId>ch.qos.logback</groupId>
+	<artifactId>logback-classic</artifactId>
+	<version>latest-logback-verison</version>
```
Comment (cschubertcs) [inline docs/java/guides/logging-overview.md:66]: Small typo
```suggestion
	<version>latest-logback-version</version>
```

Comment (cschubertcs) [reply docs/java/guides/logging-overview.md:66]: Also, scope `runtime` should probably be used here.
```suggestion
	<version>latest-logback-verison</version>
	<scope>runtime</scope>
```

Comment (MatKuhr) [reply docs/java/guides/logging-overview.md]: Yes, I was a bit unsure here. If the scope is `runtime`, would it be available when running tests?

Comment (schubertcs) [reply docs/java/guides/logging-overview.md]: Yep, it is: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope

File: docs/java/guides/logging-overview.md [hunk 4]
```diff
@@ -3,36 +3,89 @@ id: logging-overview
...
+Which artifacts are to be added exactly depends on the framework.
+
+To take Logback as an example:
+
+```xml
+<dependency>
+	<groupId>ch.qos.logback</groupId>
+	<artifactId>logback-classic</artifactId>
+	<version>latest-logback-version</version>
+    <scope>runtime</scope>
```
Comment (MatKuhr) [inline docs/java/guides/logging-overview.md:67]:
```suggestion
	<scope>runtime</scope>
```

**Pull request merged**